pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -263,17 +264,43 @@ func (c *DeploymentConfigController) reconcileDeployments(existingDeployments *k
 		}
 	}
 
-	return c.updateStatus(config)
+	return c.updateStatus(config, existingDeployments.Items)
 }
 
-func (c *DeploymentConfigController) updateStatus(config *deployapi.DeploymentConfig) error {
-	if config.Generation > config.Status.ObservedGeneration {
-		config.Status.ObservedGeneration = config.Generation
-	}
-	if _, err := c.osClient.DeploymentConfigs(config.Namespace).UpdateStatus(config); err != nil {
+func (c *DeploymentConfigController) updateStatus(config *deployapi.DeploymentConfig, deployments []kapi.ReplicationController) error {
+	newStatus, err := calculateStatus(*config, deployments)
+	if err != nil {
 		glog.V(2).Infof("Cannot update the status for %q: %v", deployutil.LabelForDeploymentConfig(config), err)
 		return transientError(fmt.Sprintf("cannot update the status for %q - requeuing", deployutil.LabelForDeploymentConfig(config)))
 	}
+
+	if !reflect.DeepEqual(*newStatus, config.Status) {
```

Comment:
`c.osClient.DeploymentConfigs(config.Namespace).UpdateStatus(config)`

The controller is a client to the apiserver as every other client.